File: 3457.eat-pizzas.java
#
# @lc app=leetcode id=3457 lang=java
#
# [3457] Eat Pizzas!
#

# @lc code=start
class Solution {
    public long maxWeight(int[] pizzas) {
        int n = pizzas.length;
        java.util.Arrays.sort(pizzas);
        int k = n / 4;
        int numOddDays = (k + 1) / 2;
        int numEvenDays = k / 2;
        long total = 0;
        for (int i = 0; i < numOddDays; i++) {
            total += pizzas[n - 1 - i];
        }
        int startPos = numOddDays;
        for (int j = 0; j < numEvenDays; j++) {
            int pos = startPos + 2 * j + 1;
            total += pizzas[n - 1 - pos];
        }
        return total;
    }
}
# @lc code=end

File: 3457.eat-pizzas.go
#
# @lc app=leetcode id=3457 lang=golang
#
# [3457] Eat Pizzas!
#

# @lc code=start
import "sort"

func maxWeight(pizzas []int) int64 {
	sort.Slice(pizzas, func(i, j int) bool {
		return pizzas[i] > pizzas[j]
	})
	k := len(pizzas) / 4
	numOdd := (k + 1) / 2
	numEven := k / 2
	ans := int64(0)
	for i := 0; i < numOdd; i++ {
		ans += int64(pizzas[i])
	}
	pos := numOdd
	for i := 0; i < numEven; i++ {
		ans += int64(pizzas[pos + 1])
		pos += 2
	}
	return ans
}
# @lc code=end